test/e2e/tests/test_trainingjob_debugger.py

```diff
@@ -0,0 +1,243 @@
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status, 
+            wait_periods,
+            period_length, 
+            get_training_resource_status, 
+            reference
+    )
+
+    def _assert_training_status_in_sync(
+        self, training_job_name, reference, expected_status
+    ):
+        assert (
+            self._wait_sagemaker_training_status(
+                training_job_name, expected_status
+            ) 
+            == self._wait_resource_training_status(reference, expected_status)
+            == expected_status
+        )
+
+    def _wait_sagemaker_training_debugger_status(
+        self, 
+        training_job_name,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            period_length,
+            get_training_debugger_sagemaker_status,
+            training_job_name,
+        )
+    
+    def _wait_resource_training_debugger_status(
+        self,
+        reference: k8s.CustomResourceReference,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status, 
+            wait_periods,
+            period_length,
+            get_training_debugger_resource_status, 
+            reference
+    )
+
+    def _assert_training_debugger_status_in_sync(
+        self, training_job_name, reference, expected_status
+    ):
+        assert (
+            self._wait_sagemaker_training_debugger_status(
+                training_job_name, expected_status
+            )
+            == self._wait_resource_training_debugger_status(reference, expected_status)
+            == expected_status
+        )
+    
+    def test_training_job_debugger(self, xgboost_training_job_debugger):
+        (reference, resource) = xgboost_training_job_debugger
+        assert k8s.get_resource_exists(reference)
+
+        training_job_name = resource["spec"].get("trainingJobName", None)
+        assert training_job_name is not None
+
+        training_job_desc = get_sagemaker_training_job(training_job_name)
+
+        assert k8s.get_resource_arn(resource) == training_job_desc["TrainingJobArn"]
+        assert training_job_desc["TrainingJobStatus"] in self.list_status_created
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "False")
+
+        self._assert_training_status_in_sync(
+            training_job_name, reference, self.status_inprogress
+        )
+
+        # Delete the k8s resource.
+        _, deleted = k8s.delete_custom_resource(reference)
+        assert deleted is True
+
+        training_job_desc = get_sagemaker_training_job(training_job_name)
+        assert training_job_desc["TrainingJobStatus"] in self.list_status_stopped
+
+    def test_completed_training_job_debugger(self, xgboost_training_job_debugger):
```

Comment:
should we just make this the test to completion for training job? do we really need a separate test?

IMO, even if we do`test_training_job_debugger` is redundant and remove canary_marker

Reply:
I will remove the canary marker from the training test, agreed. But since these are two different specs and a simple trainingjob is our most classic example, I do not want to remove it.